File: backend/app/schemas/library.py
from pydantic import BaseModel


class LibraryItem(BaseModel):
    id: str
    owner_id: str | None = None
    name: str
    type: str
    kind: str
    source: str
    size_label: str
    created_at: str


class LibraryListResponse(BaseModel):
    items: list[LibraryItem]

File: frontend/src/types/library.ts
export interface LibraryItem {
  id: string
  ownerId?: string | null
  name: string
  type: 'pdf' | 'ppt' | 'csv' | 'doc' | 'image' | 'text' | 'code'
  kind: string
  sizeLabel: string
  createdAt: string
  source: 'upload' | 'generated'
}
